integration_tests/integration/recommendations/ppcs/consecutiveSentenceDetails.spec.ts

```diff
@@ -0,0 +1,331 @@
+          ])
+
+          testConcurrentGroup(4, 3, ['sentence-seq-4-1', 'sentence-seq-4-2', 'sentence-seq-4-3'])
+
+          testSentenceSummaryDetails('sentence-seq-4-1', secondConurrentSentence, expectedLabels.offence, [
+            { key: expectedLabels.sentenceLength, value: expectedSecondTimeSentenceLength },
+          ])
+          testSentenceSummaryDetails('sentence-seq-4-2', thirdConurrentSentence, expectedLabels.offence, [
+            { key: expectedLabels.sentenceLength, value: expectedThirdTimeSentenceLength },
+          ])
+          testSentenceSummaryDetails('sentence-seq-4-3', fourthConurrentSentence, expectedLabels.offence, [
+            { key: expectedLabels.sentenceLength, value: expectedFourthTimeSentenceLength },
+          ])
+        })
+      })
+    })
+  })
+
+  const testSentenceSummaryDetails = (
+    id: string,
+    expectedSentence: PrisonSentence,
+    expectedOffenceLabel: string,
+    expectedTerms: { key: string; value: string }[]
+  ) => {
+    cy.get(`dl#${id}`).should('exist').as('sentenceSummary')
+    testSummaryList(cy.get('@sentenceSummary'), {
+      rows: {
+        matchLength: true,
+        expectedContent: [
+          { key: expectedOffenceLabel, value: expectedSentence.offences.at(0).offenceDescription },
+          { key: expectedLabels.sentenceType, value: expectedSentence.sentenceTypeDescription },
+          { key: expectedLabels.court, value: expectedSentence.courtDescription },
+          { key: expectedLabels.dateOfSentence, value: expectedSentence.sentenceDate },
+          { key: expectedLabels.startDate, value: expectedSentence.sentenceStartDate },
+          { key: expectedLabels.expiryDate, value: expectedSentence.sentenceEndDate },
+          ...expectedTerms,
+        ],
+      },
+    })
+  }
+
+  const testConsecutiveGroup = (
+    index: number,
+    titleSequenceLine: number,
+    consecutiveTo: 'index' | number,
+    expectedSummaryListIds: string[]
+  ) => {
+    cy.get(`#consecutive-group-${index}`)
+      .should('exist')
+      .should('have.class', 'govuk-inset-text')
+      .as('consecutiveGroup')
+
+    cy.get('@consecutiveGroup')
+      .find(`div#consecutive-group-title-${consecutiveTo === 'index' ? 1 : consecutiveTo}`)
+      .should('exist')
+      .as('consecutiveTitle')
+
+    cy.get('@consecutiveTitle').find('span').should('exist').should('have.text', `Sequence: line ${titleSequenceLine}`)
+    cy.get('@consecutiveTitle')
+      .find('p')
+      .should('exist')
+      .should('have.class', 'govuk-body-s')
+      .should('have.text', `Consecutive to ${consecutiveTo === 'index' ? 'index offence' : `line ${consecutiveTo}`}`)
+
+    expectedSummaryListIds.forEach(id => cy.get('@consecutiveGroup').find(`dl#${id}`).should('exist'))
+  }
+
+  const testConcurrentGroup = (
+    index: number,
+    expectedConcurrentSentenceCount: number,
+    expectedSummaryListIds: string[]
```

Comment:
Small point: since `expectedSummaryListIds` should have the same number of entries as `expectedConcurrentSentenceCount`, can we calculate the latter within this function and simplify its parameter list?